cmd/kubeadm/app/cmd/phases/controlplanepreparejoin.go

```diff
@@ -0,0 +1,226 @@
+	return workflow.Phase{
+		Name:  "control-plane-prepare",
+		Short: "Prepares the machine for serving a control plane.",
+		Long:  cmdutil.MacroCommandLongDescription,
+		Phases: []workflow.Phase{
+			newCertsSubphase(),
+			newKubeconfigSubphase(),
+			newManifestsSubphases(),
+			newCheckLocalEtcdStatusSubphase(),
+		},
+	}
+}
+
+func newCertsSubphase() workflow.Phase {
+	return workflow.Phase{
+		Name:         "certs",
+		Short:        "Generates the certificates for the new control plane components",
+		Run:          runCertsPhaseLocal,
+		InheritFlags: getControlPlanePreparePhaseFlags(),
+	}
+}
+
+func newKubeconfigSubphase() workflow.Phase {
+	return workflow.Phase{
+		Name:         "kubeconfig",
+		Short:        "Generates the kubeconfig for the new control plane components",
+		Run:          runKubeconfigPhaseLocal,
+		InheritFlags: getControlPlanePreparePhaseFlags(),
+	}
+}
+
+func newManifestsSubphases() workflow.Phase {
+	return workflow.Phase{
+		Name:  "manifests",
+		Short: "Generates the manifests for the new control plane components",
+		Phases: []workflow.Phase{
+			{
+				Name:           "all",
+				Short:          "Generates all static Pod manifest files",
+				InheritFlags:   getControlPlanePreparePhaseFlags(),
+				RunAllSiblings: true,
+			},
+			newControlPlanePrepareSubPhase(kubeadmconstants.KubeAPIServer),
+			newControlPlanePrepareSubPhase(kubeadmconstants.KubeControllerManager),
+			newControlPlanePrepareSubPhase(kubeadmconstants.KubeScheduler),
+		},
+		InheritFlags: getControlPlanePreparePhaseFlags(),
+	}
+}
+
+func newCheckLocalEtcdStatusSubphase() workflow.Phase {
+	return workflow.Phase{
+		Name:         "check-local-etcd-status",
+		Short:        "Checks the local etcd status",
+		Run:          runCheckLocalEtcdStatus,
+		InheritFlags: getControlPlanePreparePhaseFlags(),
+	}
+}
+
+func newControlPlanePrepareSubPhase(component string) workflow.Phase {
+	return workflow.Phase{
+		Name:         controlPlanePhaseProperties[component].name,
+		Short:        controlPlanePhaseProperties[component].short,
+		Run:          runControlPlaneJoinSubPhase(component),
+		InheritFlags: getControlPlanePreparePhaseFlags(),
+	}
+}
+
+func runCertsPhaseLocal(c workflow.RunData) error {
+	data, ok := c.(controlPlanePrepareData)
+	if !ok {
+		return errors.New("certs phase invoked with an invalid data struct")
+	}
+
+	// Skip if this is not a control plane
+	if data.Cfg().ControlPlane == nil {
+		return nil
+	}
+
+	cfg, err := data.InitCfg()
+	if err != nil {
+		return err
+	}
+
+	// Generate missing certificates (if any)
+	return certsphase.CreatePKIAssets(cfg)
+}
+
+func runKubeconfigPhaseLocal(c workflow.RunData) error {
+	data, ok := c.(controlPlanePrepareData)
+	if !ok {
+		return errors.New("kubeconfig phase invoked with an invalid data struct")
+	}
+
+	// Skip if this is not a control plane
+	if data.Cfg().ControlPlane == nil {
+		return nil
+	}
+
+	cfg, err := data.InitCfg()
+	if err != nil {
+		return err
+	}
+
```

Comment:
let's add something in these lines:
`fmt.Println("[control-plane-prepare] Generating kubeconfig files")`

Reply:
Every component will write `[kubeconfig] Writing %q kubeconfig file`, should we still write a generic message before those?